andorCtrl_science.py

```diff
@@ -314,25 +352,45 @@ def set_exptime(self, exptime):
 
         *WARNING* - Careful when subtracting darks as they will no longer match.
         '''
-        self.cam.GetMaximumExposure()
-        #self.pub.pprint("The maximum exposure time is %f s." % self.cam.max_exp)
+
+        self.exposure_time = exptime
+
+        # Stop the acquisition
         self.cam.AbortAcquisition()
-        time.sleep(0.5)
+
+        # Set the Read mode and the acquisition mode
+        self.cam.SetReadMode(self.ReadMode)
+        self.cam.SetAcquisitionMode(self.AcqMode)
+
+        self.cam.SetKineticCycleTime(0)
+
+        # Set the Image parameters
+        self.cam.SetIsolatedCropMode(1, self.height, self.width, self.vbin, self.hbin)
+        self.cam.SetImage(self.hbin, self.vbin, self.Lower_left_X, self.Lower_left_X+np.int(self.width)-1, self.Lower_left_Y, self.Lower_left_Y+np.int(self.height)-1)
         self.cam.SetExposureTime(exptime)
-        time.sleep(0.5)
-        self.set_video_scan()
-        time.sleep(0.2)
+        self.cam.SetShutter(0, 1, 50, 50)
+
+
+
+        self.cam.GetEMGainRange()
+        self.cam.GetEMCCDGain()
+        self.cam.GetNumberPreAmpGains()
+        self.cam.GetPreAmpGain()
+        self.cam.GetStatus()
+
+        self.cam.GetAcquisitionTimings()
+        self.pub.pprint("Actual exposure time: "+str(self.cam.exp_time))
+        self.pub.pprint("Actual accu cycle time: "+str(self.cam.accu_cycle_time))
+        self.pub.pprint("Actual kine cycle time: "+str(self.cam.kinetic_cycle_time))
+
         self.cam.StartAcquisition()
```

Comment:
@KBarjot you mean all these parameters ? In the set_exposure time ?
I added them here because when I was testing things without, there was a problem with the definition of image size...
Since there is an "AbortAcquistion" that basically kills the acquisition, I figured that we needed to re-enter all parameters of our images, instead of just the set_video_scan...

Reply:
Yes ok.